editors/code/src/commands/run_single.ts

```diff
@@ -0,0 +1,63 @@
+import * as vscode from 'vscode';
+import * as lc from 'vscode-languageclient';
+
+interface Runnable {
+    range: lc.Range;
+    label: string;
+    bin: string;
+    args: string[];
+    env: { [index: string]: string };
+}
+
+interface CargoTaskDefinition extends vscode.TaskDefinition {
+    type: 'cargo';
+    label: string;
+    command: string;
+    args: string[];
+    env?: { [key: string]: string };
+}
+
+function createTask(spec: Runnable): vscode.Task {
+    const TASK_SOURCE = 'Rust';
+    const definition: CargoTaskDefinition = {
+        type: 'cargo',
+        label: 'cargo',
+        command: spec.bin,
+        args: spec.args,
+        env: spec.env
+    };
+
+    const execOption: vscode.ShellExecutionOptions = {
+        cwd: '.',
+        env: definition.env
+    };
+    const exec = new vscode.ShellExecution(definition.command, definition.args, execOption);
+
+    const f = vscode.workspace.workspaceFolders![0];
+    const t = new vscode.Task(
+        definition,
+        f,
+        definition.label,
+        TASK_SOURCE,
+        exec,
+        ['$rustc']
+    );
+    t.presentationOptions.clear = true
+    return t;
+}
```

Comment:
Can we just import this from runnables? Or perhapse we should just keep both commands in the same file?

Reply:
I moved the handle into runnables